code/modules/power/lighting.dm

```diff
@@ -385,6 +385,67 @@
 		icon_state = "runway50"
 		base_state = "runway5"
 
+/obj/machinery/light/traffic_light
+	name = "warning light"
+	desc = "A small light used to warn when shuttle traffic is expected."
+	icon_state = "runway10"
+	base_state = "runway1"
+	fitting = "bulb"
+	brightness = 0.5
+	light_type = /obj/item/light/bulb
+	allowed_type = /obj/item/light/bulb
+	plane = PLANE_NOSHADOW_BELOW
+	on = 0
+	wallmounted = 0
+	removable_bulb = 0
+	var/connected_dock = null
+
+	New()
+		..()
+		if(src.connected_dock)
+			RegisterSignal(GLOBAL_SIGNAL, src.connected_dock, .proc/dock_signal_handler)
+
+	proc/dock_signal_handler(datum/holder, var/signal)
+		switch(signal)
+			if(COMSIG_DOCK_EVENT_INCOMING)
+				src.activate()
+			if(COMSIG_DOCK_EVENT_ARRIVED)
+				src.deactivate()
+			if(COMSIG_DOCK_EVENT_OUTGOING)
+				src.activate()
+			if(COMSIG_DOCK_EVENT_DEPARTED)
+				src.deactivate()
```

Comment:
this pattern of registering a single signal and then switching over the argument for which proc to call... hm
It's not bad per se, but it also *really* should be documented in `component_defines_global.dm` that the signal is expecting those args